dont use nogroup for networkd, resolved and timesyncd

WilliButz opened the pull request:
###### Motivation for this change
At the networkd sprint, I switched from scripted networking to networkd and stumbled over some things, one of them being that currently `systemd-networkd.service` runs as `systemd-network:nogroup`.
The upstream service file only specifies the `User` option and we didn't manually add the `systemd-network` user to the group with the same name, even though it is already defined in nixpkgs.

I noticed this when I added a wireguard interface to my configuration and wanted to pass the private key for an interface as a file, to avoid adding it to the nix store.
The following is taken from the `[Wireguard]` section of `systemd.netdev(5)`:
```
PrivateKeyFile=
           Takes an absolute path to a file which contains the Base64 encoded private key for the interface. When this option is specified, then PrivateKey= is ignored. Note that the file must be readable by the user "systemd-network", so it should be, e.g., owned by
           "root:systemd-network" with a "0640" file mode.
```

Looking at `/run/systemd/`, I noticed that this also happened for `resolved` and `timesyncd`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc #55370

WilliButz committed:
nixos/networkd: add systemd-network user to group systemd-network

WilliButz committed:
nixos/resolved: add user systemd-resolve to group systemd-resolve

WilliButz committed:
nixos/timesyncd: add user systemd-timesync to group systemd-timesync

WilliButz commented:
@GrahamcOfBorg test systemd-networkd-wireguard
@GrahamcOfBorg test systemd-timesyncd

flokli reviewed (changes requested):
@WilliButz can we update `nixos/tests/systemd-networkd-wireguard.nix` to restrict permissions of the `PrivateKeyFile` used there to that specific user?

We could use `systemd.tmpfiles.rules` with Type `f` to create the keyfile somewhere outside the nix store with restricted permissions.

That way, the nixos vm test should catch the current bug this PR would fix.

WilliButz commented:
@flokli sure, sounds good. I'll add it in about an hour :+1:

WilliButz committed:
tests/systemd-networkd: add regression test for service group

Previously systemd-networkd.service ran as systemd-network:nogroup.
The wireguard private key file is now owned by root:systemd-network with
mode 0640. It is therefore required that the systemd-network user is in the group
with the same name, so that it is able to read the key file.

WilliButz commented:
@flokli updated the test.
@GrahamcOfBorg test systemd-networkd-wireguard

(cc @NinjaTrappeur)

bluescreen303 commented:
as a consequence of this PR, I seem to now have a `resolved` user at id 1001, overlapping with a normal user account:
```
mathijs:x:1001:100:Mathijs Kwik:/home/mathijs:/run/current-system/sw/bin/zsh
resolved:x:1001:153::/var/empty:/run/current-system/sw/bin/nologin
```
